mono/mini/intrinsics.c

```diff
@@ -754,6 +754,15 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 			int dreg = alloc_preg (cfg);
 			EMIT_NEW_BIALU_IMM (cfg, ins, OP_PADD_IMM, dreg, args [0]->dreg, MONO_STRUCT_OFFSET (MonoArray, vector));
 			return ins;
+		} else if (!strcmp (cmethod->name, "GetElementCorElementType")) {
+			int preg = alloc_preg(cfg);
+			MONO_EMIT_NEW_LOAD_MEMBASE_OP_FAULT(cfg, OP_LOAD_MEMBASE, preg, args[0]->dreg, MONO_STRUCT_OFFSET(MonoObject, vtable));
+			EMIT_NEW_LOAD_MEMBASE(cfg, ins, OP_LOAD_MEMBASE, preg, preg, MONO_STRUCT_OFFSET(MonoVTable, klass));
+			EMIT_NEW_LOAD_MEMBASE(cfg, ins, OP_LOAD_MEMBASE, preg, preg, MONO_STRUCT_OFFSET(MonoClass, element_class));
+			int dreg = alloc_ireg(cfg);
+			EMIT_NEW_LOAD_MEMBASE(cfg, ins, OP_LOADI1_MEMBASE, dreg, preg, MONO_STRUCT_OFFSET(MonoClass, _byval_arg) + 10);
+			mini_type_from_op(cfg, ins, NULL, NULL);
```

Comment:
`+10` - don't know to obtain offset for https://github.com/mono/mono/blob/84f4e1dcbf96925105f4d4f54bb9cb9b61067603/mono/metadata/metadata-internals.h#L34 (offset won't work for bit fields)

Reply:
It doesn't necessarily have to be bitfield. It would be possible to make `attrs` regular `unsigned short`, `type` can be byte (or union if you want it strongly typed) and then let the compiler pack it. Not sure it's the best way.

Reply:
Would suggest adding a flag to MonoVTableFlags so it can be read from MonoVTable.flags.